test/jasmine/tests/plotschema_test.js

```diff
@@ -112,17 +113,31 @@ describe('plot schema', function() {
         expect(list).toEqual(astrs);
     });
 
-    it('layout.annotations and layout.shapes should contain `items`', function() {
-        var astrs = ['annotations', 'shapes'];
+    it('should convert _isLinkedToArray attributes to items object', function() {
+        var astrs = [
+            'annotations', 'shapes',
+            'xaxis.rangeselector.buttons', 'yaxis.rangeselector.buttons'
+        ];
 
         astrs.forEach(function(astr) {
-            expect(
-                isPlainObject(
-                    Lib.nestedProperty(
-                        plotSchema.layout.layoutAttributes, astr
-                    ).get().items
-                )
-            ).toBe(true);
+            var np = Lib.nestedProperty(
+                plotSchema.layout.layoutAttributes, astr
+            );
+
+            var name = np.parts[np.parts.length - 1],
+                itemName = name.substr(0, name.length - 1);
+
+            var itemsObj = np.get().items,
+                itemObj = itemsObj[itemName];
+
+            expect(isPlainObject(itemsObj)).toBe(true);
+            expect(itemsObj.role).toBeUndefined();
+            expect(Object.keys(itemsObj).length).toEqual(1);
+            expect(isPlainObject(itemObj)).toBe(true);
+            expect(itemObj.role).toBe('object');
+
+            var role = np.get().role;
+            expect(role).toEqual('object');
```

Comment:
@cldougl @theengineear  :arrow_double_up: needs to be satisfied for plotly.py to build to correct graph object.